components/Modal/ManageSources.js

```diff
@@ -356,100 +361,109 @@ class ManageSourcesModal extends React.Component {
         </form>
       </>
     );
-    return (
-      <Modal
-        show
-        id="cmpsr-modal-manage-sources"
-        onHide={this.props.close}
-        bsSize="large"
-        aria-labelledby="title-manage-sources"
-      >
-        <Modal.Header>
-          <Modal.CloseButton onClick={this.props.close} />
-          <Modal.Title id="title-manage-sources">
+
+    const header = (
+      <Title headingLevel="h2" size="3xl" id="title-manage-sources">
+        {(!this.state.addEntry && (
+          <FormattedMessage
+            defaultMessage="Sources"
+            description="Sources provide the contents from which components are selected"
+          />
+        )) ||
+          (this.state.editName === "" && <FormattedMessage defaultMessage="Add source" />) || (
+            <FormattedMessage defaultMessage="Edit source" />
+          )}
+      </Title>
+    );
+
+    const body = (
```

Comment:
phew, this contains a lot of unnecessary brackets and fragments. It is not directly in return of render, thus you can just write something like:
```
const body = Object.keys(manageSources.sources).length === 0 ?              
          <EmptyState                                                           
            title={formatMessage(messages.errorStateTitle)}                     
            message={formatMessage(messages.errorStateMessage)}                 
          />                                                                    
        : !this.state.addEntry ? 
           <> 
               ...
```

Same for `footer`, But mostly just preference of style, so feel free to ignore :)

Reply:
Refactored for better style